src/main/java/uk/gov/hmcts/reform/civil/service/StateGeneratorService.java

```diff
@@ -31,15 +35,22 @@ public CaseState getCaseStateForEndJudgeBusinessProcess(CaseData data) {
             decision = null;
         }
 
-        if (decision == MAKE_AN_ORDER && !isBlank(directionsText)) {
+        if (Boolean.TRUE.equals(hasDismissedCase)) {
+            return APPLICATION_DISMISSED;
+        } else if (decision == MAKE_AN_ORDER && !isBlank(directionsText)) {
             return AWAITING_DIRECTIONS_ORDER_DOCS;
-        }
-        if (decision == REQUEST_MORE_INFO) {
+        } else if (decision == REQUEST_MORE_INFO) {
             return AWAITING_ADDITIONAL_INFORMATION;
-        }
-        if (decision == MAKE_ORDER_FOR_WRITTEN_REPRESENTATIONS) {
+        } else if (decision == MAKE_ORDER_FOR_WRITTEN_REPRESENTATIONS) {
             return AWAITING_WRITTEN_REPRESENTATIONS;
         }
         return data.getCcdState();
     }
+
+    private boolean hasCaseDismissed(GAJudicialMakeAnOrder gaJudicialMakeAnOrder) {
```

Comment:
Just suggestion... as method argument would take whole `CaseData` object for readability purpose. Then create descriptive variables name for better understand what this method does.

Reply:
perfect.
Thanks.